### PR TITLE
feat(pattern/input): add small input text

### PR DESCRIPTION
**Pull request by tiloyi**
## I have read [the contributing guidelines](https://mozaic.adeo.cloud/Contributing/)

- [x] Yes
- [ ] No

## Does this PR introduce a breaking change?

read [What is a breaking change ?](https://mozaic.adeo.cloud/Contributing/Prerequisite/GitConventions/#breaking-changes-)

- [ ] Yes
- [x] No

## Describe the changes

Pattern : Add new size for input fields

Github issue number or Jira issue URL: **MZC-164**

### CODE REVIEW

File: src/docs/Components/TextInput/code.mdx [hunk 1]
```diff
@@ -38,6 +38,17 @@ To create an input you have one main class to apply `mc-text-input`.
...
+## Variations
+
+### Available sizes
+
+You can use one of the 2 available sizes :
+
+- **small** : `mc-text-input--s`
+- **medium** : this is the default style so you don't need to add a modifier class
+
+<Preview path="input-sizes" />
```
Comment (gael-boyenval) [inline src/docs/Components/TextInput/code.mdx]: should we use a responsive modifier ? 
could an input be small on mobile and medium on other devices ?

File: packages/styles/settings-tools/_s.text-input.scss [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  $padding-h: map-get($sizeProps, 'padding-h');
+  $padding-t: map-get($sizeProps, 'padding-t');
+
+  @include set-font-scale(
+    map-get($sizeProps, 'font-size'),
+    map-get($sizeProps, 'line-height')
+  );
+
+  // remove border height and center baseline optically
+  padding:
```
Comment (gael-boyenval) [inline packages/styles/settings-tools/_s.text-input.scss]: design : the text does not appear to be optically centered in the small variation, you may need another calculation

File: src/docs/Components/TextInput/code.mdx (hunk 1 above)
Comment (nherwegh) [reply src/docs/Components/TextInput/code.mdx]: yes I think some designers will want to do something like that

File: src/docs/Components/TextInput/index.mdx [hunk 3]
```diff
@@ -13,6 +13,12 @@ status:
...
 
+## Variations
+
+We have two different sizes of text inputs to answer differents needs. You can use both cases to suit your interface the best. Just keep in mind touch devices when placing smaller elements near each others.
+
+<Preview path="input-sizes" />
+
 ## Input with icon :
 
 Use a left icon to inform about a specific information type or formatting you expect from the user.
```
Comment (caqueste) [inline src/docs/Components/TextInput/index.mdx]: I think it's better to structure like this :
## Variations (##)
### Sizes (###)
### Input with icon (###)

File: src/docs/Components/TextInput/index.mdx [hunk 4]
```diff
@@ -30,3 +36,7 @@ Use a left icon to inform about a specific information type or formatting you ex
 <HintItem>
   Always use a label associated with an input, even using placeholders
 </HintItem>
+
+<HintItem dont>
```
Comment (caqueste) [inline src/docs/Components/TextInput/index.mdx]: Can we add a "don't" to specity this: Don't use a medium input with a small button.
What do you think?

File: src/docs/Components/TextInput/code.mdx [hunk 5]
```diff
@@ -38,6 +38,17 @@ To create an input you have one main class to apply `mc-text-input`.
 
 <Preview path="input-default" />
 
+## Variations
+
+### Available sizes
+
+You can use one of the 2 available sizes :
+
+- **small** : `mc-text-input--s`
```
Comment (caqueste) [inline src/docs/Components/TextInput/code.mdx]: ~small~ Small

File: src/docs/Components/TextInput/code.mdx [hunk 6]
```diff
@@ -38,6 +38,17 @@ To create an input you have one main class to apply `mc-text-input`.
...
+- **medium** : this is the default style so you don't need to add a modifier class
```
Comment (caqueste) [inline src/docs/Components/TextInput/code.mdx]: ~medium~ Medium

File: src/docs/Components/TextInput/code.mdx (hunk 1 above)
Comment (caqueste) [reply src/docs/Components/TextInput/code.mdx]: @tiloyi The small input has a height of 2mu (32px), on the preview it make 34px.
The baseline of the label is centered with the height of the input :
Capture d’écran 2019-12-04 à 11 52 58

File: src/docs/Components/TextInput/index.mdx [hunk 7]
```diff
@@ -13,6 +13,12 @@ status:
 
 <Preview nude path="input-state" />
 
+## Variations
+
+We have two different sizes of text inputs to answer differents needs. You can use both cases to suit your interface the best. Just keep in mind touch devices when placing smaller elements near each others.
+
+<Preview path="input-sizes" />
```
Comment (caqueste) [inline src/docs/Components/TextInput/index.mdx]: @tiloyi The small input has a height of 2mu (32px), on the preview it make 34px.
The baseline of the label is centered with the height of the input :
Capture d’écran 2019-12-04 à 11 52 58

Comment (tiloyi) [reply src/docs/Components/TextInput/index.mdx]: Fixed.

File: src/docs/Components/TextInput/code.mdx (hunk 6 above)
Comment (tiloyi) [reply src/docs/Components/TextInput/code.mdx]: fixed

File: src/docs/Components/TextInput/code.mdx (hunk 5 above)
Comment (tiloyi) [reply src/docs/Components/TextInput/code.mdx]: fixed

File: src/docs/Components/TextInput/index.mdx (hunk 3 above)
Comment (tiloyi) [reply src/docs/Components/TextInput/index.mdx]: fixed

File: src/docs/Components/TextInput/index.mdx (hunk 4 above)
Comment (tiloyi) [reply src/docs/Components/TextInput/index.mdx]: fixed

File: packages/styles/components/_c.text-input.scss [hunk 8]
```diff
@@ -13,11 +11,6 @@
   background-color: $color-input-background;
   border-radius: get-border-radius('m');
 
-  // remove border height and center baseline optically
-  padding:
-    calc(#{$mu075} - 0.125em - #{$border-width}) $mu075
-    calc(#{$mu075} + 0.125em - #{$border-width});
-
   &::placeholder {
     @include set-font-scale('05');
```
Comment (gael-boyenval) [inline packages/styles/components/_c.text-input.scss]: Put the placeholder inside your mixin and define the same font-size and line-height for it as the value. 
on safari this is what it look like :
Capture d’écran 2019-12-04 à 18 14 10

Comment (gael-boyenval) [reply packages/styles/components/_c.text-input.scss]: If it is not the source of the problem, I will pull your branch to deboug this.

File: src/docs/Components/TextInput/code.mdx (hunk 1 above)
Comment (tiloyi) [reply src/docs/Components/TextInput/code.mdx]: fixed

File: packages/styles/components/_c.text-input.scss (hunk 8 above)
Comment (tiloyi) [reply packages/styles/components/_c.text-input.scss]: I make the modification, but I can't test it, 'cause it's a Safari issue.
Can you check on your side and tell me if it's okay?

Comment (tiloyi) [reply packages/styles/components/_c.text-input.scss]: @gael-boyenval about the line-height, it's fixed, but about the icon-size the designers decided to leave it like that.

**Pull request merged**